agent/handlers/types/v2/response.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package v2
+
+import (
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/containermetadata"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+// TaskResponse defines the schema for the task response JSON object
+type TaskResponse struct {
+	Cluster       string
+	TaskARN       string
+	Family        string
+	Version       string
+	DesiredStatus string `json:",omitempty"`
+	KnownStatus   string
+	Containers    []ContainerResponse `json:",omitempty"`
+	Limits        LimitsResponse      `json:",omitempty"`
+}
+
+// ContainerResponse defines the schema for the container response
+// JSON object
+type ContainerResponse struct {
+	ID            string `json:"DockerId"`
+	Name          string
+	DockerName    string
+	Image         string
+	ImageID       string
+	Ports         []PortResponse    `json:",omitempty"`
+	Labels        map[string]string `json:",omitempty"`
+	DesiredStatus string
+	KnownStatus   string
+	ExitCode      *int `json:",omitempty"`
+	Limits        LimitsResponse
+	CreatedAt     *time.Time `json:",omitempty"`
+	StartedAt     *time.Time `json:",omitempty"`
+	FinishedAt    *time.Time `json:",omitempty"`
+	Type          string
+	Networks      []containermetadata.Network `json:",omitempty"`
+}
+
+// LimitsResponse defines the schema for task/cpu limits response
```

Comment:
This comment seems a bit confusing. is it the schema for cpu/memory limits?

Reply:
correct